### ISSUE TITLE
Add option to Backup Profile(Settings, History, Paswords, etc) to a file and then restore when required.

### ISSUE THREAD

**Issue by TontyTon** [brave/brave-browser#4062]
## Description 

Add option to Backup Profile to a file and then restore when required.
All these things can be backed up -

- Settings
- History
- Bookmarks
- Passwords (Important)
- Search Engines
- Autofill, etc.
And extensions and their settings (if possible)

This will be very useful when we have to set up multiple PCs
(copying the profile folder doesn't work, only few of the above things sustain)

This can also solve the issue https://github.com/brave/brave-browser/issues/1357 about switching between release, beta, dev and nighly versions

## Steps to Reproduce

   1. 
   2. 
   3. 

## Expected result:
Able to backup all things listed and then restore then

**Comment by Quisher** [comment]
Was going to do a fresh install on my pc and realized there is no way to keep my brave browsing passwords etc. Kinda feeling I will end up wasting all the months of using brave as my primary browser.

Is it there on the roadmap even? As I see hasn't been assigned yet.

**Comment by TontyTon** [comment]
Will this feature be added to Brave???

**Comment by technodrome** [comment]
By the way, it's 2020. Not having a possibility to back up the whole user profile is something one might expect in 1996, not in this era.

**Comment by bsclifton** [comment]
While there isn't a way to export this in the UI, this data is being stored in a location that is quite easy to back up 😄 

- On Windows, you can find the profiles under `%USERPROFILE%\AppData\Local\BraveSoftware\`
- On macOS, you can find the profiles under `~/Library/Application Support/BraveSoftware/`
- On Linux, you can find the profiles under `~/.config/BraveSoftware/`
- Depending on the channel (Release, Beta, Dev, Nightly) the profile directory will be named differently. 
    - Release = `Brave-Browser`
    - Beta = `Brave-Browser-Beta`
    - Dev = `Brave-Browser-Dev`
    - Nightly = `Brave-Browser-Nightly`

You can basically zip this profile and use that as a backup. 

If you needed to jump between channels, you can do this by renaming the profile directory. Please be careful doing this though as there can be problems if the version you're trying to move to has a lower version of Chromium than the profile you're trying to move. For example, Nightly is currently on Chromium 81 while Release and Beta are on Chromium 80. Moving from Release or Beta to Nightly is OK, but not the other way around (until Chromium is the same)

Hope this helps 😄 I agree, having a way in the UI to click a button and then be presented with a .zip file that you can download / save somewhere would be a great experience

**Comment by TontyTon** [comment]
No, it don't work most of the times, something or other get removed. Even when moving from release to beta not work sometimes. Sometimes passwords, sometimes rewards account(restore don't work either, account get restored, but balance remains 0)...
Yeah it do work in same channels, but it takes a lot of space. The thing which can come under 20 MBs take space of more than 300MBs.

**Comment by danrobi11** [comment]
Yes, export option is needed

**Comment by delfick** [comment]
@bsclifton I did that recently (on a Mac) but when I copy it back into place and open Brave it tells me the settings are corrupt and resets them and so I have to try and remember what extensions and settings I had and redo them. Is there something we need to do to that folder to make Brave trust it and use those settings?

**Comment by technodrome** [comment]
2 years later and not a single beep from the devs? Seems like best time to change browsers, then.

**Comment by lhdgod** [comment]
My extensions, Twitter, Gmail account don't survive the transfer. Afaik, It relates to encrypted Machine Hardware ID or something like that.

**Comment by SaintZ13** [comment]
3 years later and this is still an issue and not a single word from the developers... terrible

**Comment by TontyTon** [comment]
Just putting out some links here which may be useful who want to backup and restore, 
1) https://superuser.com/questions/951095/
2) https://superuser.com/questions/1646052/


Some info in this answer might be useful to reduce backup size (though, old post) - https://superuser.com/a/14157

**Comment by Neustradamus** [comment]
Yes, it is missing.
